src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -296,3 +301,35 @@ AsyncLogWriter::BufferUpdater::~BufferUpdater() {
     p->_buffer_staging = _buf2;
   }
 }
+
+bool AsyncLogWriter::enqueue_if_initialized(LogFileStreamOutput& output,
+                                            const LogDecorations& decorations, const char* msg) {
+  AsyncLogWriter* instance = AsyncLogWriter::instance();
+  if (instance != nullptr) {
+    if ((uintptr_t)instance == (uintptr_t)Thread::current_or_null()) {
```

Comment:
Why the casts ??

Reply:
Sorry, that was sloppy of me. I still need a cast, but to the proper `Thread` type.

Reply:
Why do you need a cast to `Thread`? `AsyncLogWriter` is-a NonJavaThread is-a `Thread`. `Thread::current_or_null()` returns a `Thread`.